src/java/org/apache/cassandra/index/sai/disk/v1/bitpack/NumericValuesWriter.java

```diff
@@ -22,79 +22,98 @@
 
 import javax.annotation.concurrent.NotThreadSafe;
 
+import com.google.common.annotations.VisibleForTesting;
+
 import org.apache.cassandra.index.sai.disk.format.IndexComponent;
 import org.apache.cassandra.index.sai.disk.format.IndexDescriptor;
 import org.apache.cassandra.index.sai.disk.v1.MetadataWriter;
 import org.apache.cassandra.index.sai.disk.v1.SAICodecUtils;
 import org.apache.lucene.store.IndexOutput;
 
 @NotThreadSafe
-public class NumericValuesWriter implements Closeable
+public interface NumericValuesWriter extends Closeable
 {
-    public static final int MONOTONIC_BLOCK_SIZE = 16384;
-    public static final int BLOCK_SIZE = 128;
-
-    private final IndexOutput output;
-    private final AbstractBlockPackedWriter writer;
-    private final MetadataWriter metadataWriter;
-    private final String componentName;
-    private final int blockSize;
-    private long count = 0;
-
-    public NumericValuesWriter(String componentName,
-                               IndexOutput indexOutput,
-                               MetadataWriter metadataWriter,
-                               boolean monotonic) throws IOException
+    NumericValuesWriter NOOP_WRITER = new NumericValuesWriter()
```

Comment:
This doesn't seem used anywhere.

Reply:
hmmm, yeah, the interface probably collapses if nothing actually needs it

Reply:
This was put in when we were only, optionally, creating the partition size structure but isn't needed any more. I'll remove it and get rid of it and the interface.